sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5113,7 +5113,7 @@ object SQLConf {
         "the sequence of steps that the query performs in a composable fashion.")
       .version("4.0.0")
       .booleanConf
-      .createWithDefault(Utils.isTesting)
+      .createWithDefault(true)
```

Comment:
We need test cases to check when the configuration is false.

Reply:
Sounds good, I added test coverage to exercise the error message with pipe syntax when the config is disabled.